File: backend/src/main/java/com/yagubogu/checkin/dto/FanRateResponse.java
package com.yagubogu.checkin.dto;

import java.util.ArrayList;
import java.util.List;

public record FanRateResponse(
        List<FanRateByGameResponse> fanRateByGames
) {

    public static FanRateResponse from(
            FanRateByGameResponse myTeamResponse,
            List<FanRateGameEntry> exceptMyTeam
    ) {
        List<FanRateByGameResponse> fanRateByGameResponses = new ArrayList<>();
        fanRateByGameResponses.add(myTeamResponse);

        List<FanRateByGameResponse> sortedOthers = exceptMyTeam.stream()
                .map(FanRateGameEntry::getResponse)
                .sorted()
                .toList();

        fanRateByGameResponses.addAll(sortedOthers);

        return new FanRateResponse(fanRateByGameResponses);
    }
}

File: android/app/src/main/java/com/yagubogu/data/dto/response/checkin/FanRateResponse.kt
package com.yagubogu.data.dto.response.checkin

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class FanRateResponse(
    @SerialName("fanRateByGames")
    val fanRateByGames: List<FanRateByGameDto>,
)
